### PR TITLE
Rename `Arguments` to `Parameters` in the AST

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

This PR renames a few AST nodes for clarity:

- `Arguments` is now `Parameters`
- `Arg` is now `Parameter`
- `ArgWithDefault` is now `ParameterWithDefault`

For now, the attribute names that reference `Parameters` directly are changed (e.g., on `StmtFunctionDef`), but the attributes on `Parameters` itself are not (e.g., `vararg`). We may revisit that decision in the future.

For context, the AST node formerly known as `Arguments` is used in function definitions. Formally (outside of the Python context), "arguments" typically refers to "the values passed to a function", while "parameters" typically refers to "the variables used in a function definition". E.g., if you Google "arguments vs parameters", you'll get some explanation like:

> A parameter is a variable in a function definition. It is a placeholder and hence does not have a concrete value. An argument is a value passed during function invocation.

We're thus deviating from Python's nomenclature in favor of a scheme that we find to be more precise.

### CODE REVIEW

File: crates/ruff_python_ast/src/nodes.rs [hunk 1]
```diff
@@ -1822,22 +1822,9 @@ impl From<ExceptHandlerExceptHandler> for ExceptHandler {
     }
 }
 
-/// See also [arguments](https://docs.python.org/3/library/ast.html#ast.arguments)
-#[derive(Clone, Debug, PartialEq)]
-pub struct PythonArguments {
```
Comment (charliermarsh) [inline crates/ruff_python_ast/src/nodes.rs:1827]: I removed this -- I don't think it's used anymore?

File: crates/ruff_python_ast/src/node.rs [hunk 2]
```diff
@@ -3661,9 +3661,9 @@ pub enum AnyNodeRef<'a> {
     PatternMatchAs(&'a ast::PatternMatchAs),
     PatternMatchOr(&'a ast::PatternMatchOr),
     Comprehension(&'a Comprehension),
-    Arguments(&'a Arguments),
-    Arg(&'a Arg),
-    ArgWithDefault(&'a ArgWithDefault),
+    Arguments(&'a Parameters),
+    Arg(&'a Parameter),
+    ArgWithDefault(&'a ParameterWithDefault),
```
Comment (MichaReiser) [inline crates/ruff_python_ast/src/node.rs:3666]: Rename

File: crates/ruff_python_ast/src/node.rs [hunk 3]
```diff
@@ -90,9 +90,9 @@ pub enum AnyNode {
     PatternMatchAs(ast::PatternMatchAs),
     PatternMatchOr(ast::PatternMatchOr),
     Comprehension(Comprehension),
-    Arguments(Arguments),
-    Arg(Arg),
-    ArgWithDefault(ArgWithDefault),
+    Arguments(Parameters),
+    Arg(Parameter),
+    ArgWithDefault(ParameterWithDefault),
```
Comment (MichaReiser) [inline crates/ruff_python_ast/src/node.rs:95]: Rename the variants to match the new node names

File: crates/ruff_python_ast/src/nodes.rs [hunk 4]
```diff
@@ -140,7 +140,7 @@ impl From<StmtFunctionDef> for Stmt {
 pub struct StmtAsyncFunctionDef {
     pub range: TextRange,
     pub name: Identifier,
-    pub args: Box<Arguments>,
+    pub args: Box<Parameters>,
```
Comment (MichaReiser) [inline crates/ruff_python_ast/src/nodes.rs:143]: Should we rename the field to `parameters`?

**Review by MichaReiser — APPROVED** [review]

**Review by zanieb — APPROVED** [review]
I read most of this :)

**Pull request merged**